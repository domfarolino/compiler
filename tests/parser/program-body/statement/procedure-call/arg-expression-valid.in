program dominic is
  procedure fn(integer arg1 in, float arg2 in, bool arg3 in, string arg4 out, char arg5 inout)
  begin
  end procedure;

  string arg4;
  char arg5;

begin
  fn(10, (11.11 * 12) + 13, (10 == 4858), arg4, arg5);
end program.
