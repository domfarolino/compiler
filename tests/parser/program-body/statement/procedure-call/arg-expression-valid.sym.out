+--------Symbol table--------+
     char            arg5          isGlobal:     -- paramType: InOut
   string            arg4          isGlobal:     -- paramType:   Out
    float            arg2          isGlobal:     -- paramType:    In
     bool            arg3          isGlobal:     -- paramType:    In
  integer            arg1          isGlobal:     -- paramType:    In
procedure              fn          isGlobal:     -- paramType:  None
+--------End symbol table----+

+--------Symbol table--------+
     char            arg5          isGlobal:     -- paramType:  None
   string            arg4          isGlobal:     -- paramType:  None
procedure              fn          isGlobal:     -- paramType:  None
+--------End symbol table----+

Program compiled successfully
[1;31m[0m